pkg/ingestor/key/key.go

```diff
@@ -0,0 +1,115 @@
+//
```

Comment:
Do we want to store any hints about the key types at all, or would the expectation be that it is always just a collection of bytes and we have to figure out what key type it is?